### PR TITLE
Color mapping features from leaflet and rasterfaster

### PR DESCRIPTION
**Pull request by jcheng5**
Adds `col_numeric`, `col_bin`, `col_quantile`, `col_factor`, and `colour_ramp` functions.

### CODE REVIEW

**Comment by jcheng5** [comment]
Note that this brings in Rcpp and RcppParallel, the latter of which causes these kinds of warnings: https://github.com/RcppCore/RcppParallel/issues/5

I can easily switch to non-parallel if you prefer, and drop the RcppParallel requirement. The parallel is faster but not markedly so (maybe 2X I think).

**Comment by jcheng5** [comment]
Oh, and camel case to underscore. OK, I have some work to do...

File: R/colour-mapping.r [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+#'   domain.
+#' @param na.color The color to return for \code{NA} values. Note that
+#'   \code{na.color=NA} is valid.
+#'
+#' @return A function that takes a single parameter \code{x}; when called with a
+#'   vector of numbers (except for \code{col_factor}, which expects
+#'   factors/characters), #RRGGBB color strings are returned.
+#'
+#' @export
+col_numeric = function(palette, domain, na.color = "#808080") {
```
Comment (hadley) [inline R/colour-mapping.r]: Can you use `<-` for assignment please?

Comment (jcheng5) [inline R/colour-mapping.r]: Done.

**Comment by jcheng5** [comment]
@hadley Do you mind taking a look at this when you get a chance? We want to take Leaflet to CRAN soon and we need to decide whether we want a dependency on scales or rasterfaster.

Thanks!

**Comment by hadley** [comment]
LGTM.  Do you want me to plan a scales release?

**Comment by jcheng5** [comment]
That would be great, thanks--I can basically release Leaflet once scales is
up on CRAN.

**Pull request merged**